### PR TITLE
Ci publish template

### PR DESCRIPTION
**Pull request by aminya**
The publish template cherry-picked from https://github.com/atom/atom/pull/21081.

See that pull request for more information.

### CODE REVIEW

File: script/vsts/platforms/templates/publish.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+parameters:
+  # artifacts:
+  # - fileName
+  # - fileDir
+  # - condition
+  # - continueOnError
```
Comment (sadick254) [inline script/vsts/platforms/templates/publish.yml:6]: Is there a reason why these are commented out?

Comment (aminya) [reply script/vsts/platforms/templates/publish.yml:6]: This is the API documentation. It is not code.

Comment (aminya) [reply script/vsts/platforms/templates/publish.yml]: I changed the explanation in https://github.com/atom/atom/pull/21262/commits/366c3b56f6e8da86664baa6c5ef373e142e269a7

**Review by sadick254 — APPROVED** [review]
This looks great. 👍

**Pull request merged**